src/Features/Core/Portable/PdbSourceDocument/PdbSourceDocumentLoaderService.cs

```diff
@@ -105,14 +109,14 @@ public PdbSourceDocumentLoaderService([Import(AllowDefault = true)] ISourceLinkS
 
         private async Task<SourceFileInfo?> TryGetSourceLinkFileAsync(SourceDocument sourceDocument, Encoding encoding, IPdbSourceDocumentLogger? logger, CancellationToken cancellationToken)
         {
-            if (_sourceLinkService is null || sourceDocument.SourceLinkUrl is null)
+            if (sourceDocument.SourceLinkUrl is null || _sourceLinkService.Value is null)
                 return null;
 
             // This should ideally be the repo-relative path to the file, and come from SourceLink: https://github.com/dotnet/sourcelink/pull/699
             var relativePath = Path.GetFileName(sourceDocument.FilePath);
 
             var delay = Task.Delay(SourceLinkTimeout, cancellationToken);
-            var sourceFileTask = _sourceLinkService.GetSourceFilePathAsync(sourceDocument.SourceLinkUrl, relativePath, logger, cancellationToken);
+            var sourceFileTask = _sourceLinkService.Value.GetSourceFilePathAsync(sourceDocument.SourceLinkUrl, relativePath, logger, cancellationToken);
```

Comment:
Can/should this do a switch to the background thread to ensure the load happens there?

Reply:
(although I guess any thread switch should probably be in the implementation since this code doesn't know anything about threads like that...)

Reply:
Maybe. I'll leave that for @davidwengier . This is just me doing my best to play whack-a-mole on RPS regressions because of a new DLL load. Thread wouldn't reduce that problem :/

Reply:
>  This is just me doing my best to play whack-a-mole on RPS regressions

And I'm very grateful because I have no idea why the first commit wouldn't have solved the problem ¯\\\_(ツ)_/¯